docs/book/component-guide/orchestrators/modal.md

```diff
@@ -0,0 +1,375 @@
+{% endhint %}
+
+### Additional configuration
+
+Here's how to configure both types of settings:
+
+```python
+from zenml.integrations.modal.flavors.modal_orchestrator_flavor import (
+    ModalOrchestratorSettings
+)
+from zenml.config import ResourceSettings
+
+# Configure Modal-specific settings
+modal_settings = ModalOrchestratorSettings(
+    gpu="A100",                # GPU type (optional)
+    region="us-east-1",        # Preferred region
+    cloud="aws",               # Cloud provider
+    execution_mode="pipeline", # or "per_step"
+    timeout=3600,              # 1 hour timeout
+    min_containers=1,          # Keep warm containers
+    max_containers=10,         # Scale up to 10 containers
+)
+
+# Configure hardware resources (quantities)
+resource_settings = ResourceSettings(
+    cpu_count=16,              # Number of CPU cores
+    memory="32GB",             # 32GB RAM
+    gpu_count=1                # Number of GPUs (combined with gpu type below)
+)
+
+@pipeline(
+    settings={
+        "orchestrator": modal_settings,
+        "resources": resource_settings
+    }
+)
+def my_modal_pipeline():
+    # Your pipeline steps here
+    ...
+```
+
+### Resource configuration
+
+{% hint style="info" %}
+**Pipeline-Level Resources**: The Modal orchestrator uses pipeline-level resource settings to configure the Modal function for the entire pipeline. All steps share the same Modal function resources. Configure resources at the `@pipeline` level for best results.
+{% endhint %}
+
+You can configure pipeline-wide resource requirements using `ResourceSettings` for hardware resources and `ModalOrchestratorSettings` for Modal-specific configurations:
+
+```python
+from zenml.config import ResourceSettings
+from zenml.integrations.modal.flavors.modal_orchestrator_flavor import (
+    ModalOrchestratorSettings
+)
+
+# Configure resources at the pipeline level (recommended)
+@pipeline(
+    settings={
+        "resources": ResourceSettings(
+            cpu_count=16,
+            memory="32GB", 
+            gpu_count=1        # These resources apply to the entire pipeline
+        ),
+        "orchestrator": ModalOrchestratorSettings(
+            gpu="A100",        # GPU type for the entire pipeline
+            region="us-west-2"
+        )
+    }
+)
+def my_pipeline():
+    first_step()   # Runs with pipeline resources: 16 CPU, 32GB RAM, 1x A100
+    second_step()  # Runs with same resources: 16 CPU, 32GB RAM, 1x A100
+    ...
+
+@step
+def first_step():
+    # Uses pipeline-level resource configuration
+    ...
+
+@step
+def second_step():
+    # Uses same pipeline-level resource configuration
+    ...
+```
+
+### Execution modes
+
+The Modal orchestrator supports two execution modes:
+
+1. **`pipeline` (default)**: Runs the entire pipeline in a single Modal function for maximum speed and cost efficiency
```

Comment:
Not sure I understand why this `pipeline` option is max speed. Isn't it running everything sequentially in the same container? Wouldn't running things in parallel in separate Modal function calls run faster?